File: design-patterns/go/template_method/transport/car.go
// car.go

package main

import "fmt"

type Car struct {
	*Transport
}

func NewCar() (car *Car) {
	car = &Car{}
	car.Transport = NewTransport(car)
	return
}

func (car *Car) AddEngine() {
	fmt.Println("Adding Engine to Car")
}

func (car *Car) AddWheel() {
	fmt.Println("Adding 4 Wheels to Car")
}

func (car *Car) AddWing() {
	// implementation not require for car
}

func (car *Car) CreateBody() {
	fmt.Println("Creating Car Body")
}

File: design-patterns/php/templatemethod/transport/Car.php
<?php
// Car.php

namespace BigBoxCode\DesignPattern\TemplateMethod\Transport;


class Car extends Transport {
    public function createBody(): void {
        echo "Creating Car Body\n";
    }

    public function addEngine(): void {
        echo "Adding Engine to Car\n";
    }

    public function addWheel(): void {
        echo "Adding 4 Wheels to Car\n";
    }

    public function addWing(): void {
        // Not required for Car
    }
}